app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java

```diff
@@ -0,0 +1,183 @@
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.RectF;
+
+import com.github.mikephil.charting.animation.ChartAnimator;
+import com.github.mikephil.charting.buffer.BarBuffer;
+import com.github.mikephil.charting.data.BarData;
+import com.github.mikephil.charting.data.BarEntry;
+import com.github.mikephil.charting.highlight.Highlight;
+import com.github.mikephil.charting.highlight.Range;
+import com.github.mikephil.charting.interfaces.dataprovider.BarDataProvider;
+import com.github.mikephil.charting.interfaces.datasets.IBarDataSet;
+import com.github.mikephil.charting.renderer.BarChartRenderer;
+import com.github.mikephil.charting.utils.Transformer;
+import com.github.mikephil.charting.utils.Utils;
+import com.github.mikephil.charting.utils.ViewPortHandler;
+import com.github.mikephil.charting.model.GradientColor;
+import android.graphics.LinearGradient;
+
+public class BarChartRoundedRenderer extends BarChartRenderer {
+    private static final float CORNER_RADIUS = 10;
+
+    BarChartRoundedRenderer(BarDataProvider chart, ChartAnimator animator,
+            ViewPortHandler viewPortHandler) {
+        super(chart, animator, viewPortHandler);
+    }
+
+    private void drawRectHelper(Canvas c, float left, float top, float right, float bottom, Paint p) {
+        if (bottom - top >= CORNER_RADIUS) {
+            c.drawRoundRect(left, top, right, bottom, CORNER_RADIUS, CORNER_RADIUS, p);
+            c.drawRect(left, top+CORNER_RADIUS, right, bottom, p);
+        }
+        else {
+            c.drawRoundRect(left, top, right, bottom, CORNER_RADIUS, CORNER_RADIUS, p);
+            c.drawRect(left, top + (bottom - top)*3/4, right, bottom, p);
+        }
+    }
+
+    private void drawRectHelper(Canvas c, RectF r, Paint p) {
+        drawRectHelper(c, r.left, r.top, r.right, r.bottom, p);
+    }
+
+    private RectF mBarShadowRectBuffer = new RectF();
+
+    @Override
+    public void drawDataSet(Canvas c, IBarDataSet dataSet, int index) {
+
+        Transformer trans = mChart.getTransformer(dataSet.getAxisDependency());
+        mBarBorderPaint.setColor(dataSet.getBarBorderColor());
+        mBarBorderPaint.setStrokeWidth(Utils.convertDpToPixel(dataSet.getBarBorderWidth()));
+        final boolean drawBorder = dataSet.getBarBorderWidth() > 0.f;
+        float phaseX = mAnimator.getPhaseX();
+        float phaseY = mAnimator.getPhaseY();
+        // draw the bar shadow before the values
+        if (mChart.isDrawBarShadowEnabled()) {
+            mShadowPaint.setColor(dataSet.getBarShadowColor());
+
+            BarData barData = mChart.getBarData();
+            final float barWidth = barData.getBarWidth();
+            final float barWidthHalf = barWidth / 2.0f;
+            float x;
+            int count = count = Math.min((int)(Math.ceil((float)(dataSet.getEntryCount()) * phaseX)), dataSet.getEntryCount());
+            for (int i = 0; i < count; i++) {
```

Comment:
add a comment for this for loop

Reply:
Deleted whole section we don't use shadows